packages/opentelemetry-core/src/trace/sampler/ParentOrElseSampler.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Sampler,
+  SpanContext,
+  TraceFlags,
+  SamplingDecision,
+  SpanKind,
+  Attributes,
+  Link,
+} from '@opentelemetry/api';
+
+/** Sampler that samples a given fraction of traces. */
+export class ParentOrElseSampler implements Sampler {
+  constructor(private _delegateSampler: Sampler) {}
+
+  shouldSample(
+    parentContext: SpanContext | undefined,
+    traceId: string,
+    spanName: string,
+    spanKind: SpanKind,
+    attributes: Attributes,
+    links: Link[]
+  ) {
+    // Respect the parent sampling decision if there is one
+    if (parentContext && typeof parentContext.traceFlags !== 'undefined') {
```

Comment:
As mentioned by @kintel https://github.com/open-telemetry/opentelemetry-js/commit/5228dfa6cfed1e188608a419c5c895c34b425400#r40332952

The traceFlags are not optional, so will never be undefined.